protos/chaincode.proto

```diff
@@ -94,7 +94,7 @@ message ChaincodeInvocationSpec {
 
     ChaincodeSpec chaincodeSpec = 1;
     //ChaincodeInput message = 2;
-
+    string userGivenID = 2;
```

Comment:
seems this field is a good place for a comment explaining rationale for this work. Also perhaps change ID to TID or TransID (perhaps shorten to givenTransID or useTransID or ... )?